exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillResultSetImpl.java

```diff
@@ -98,6 +102,32 @@ private void throwIfClosed() throws AlreadyClosedSqlException,
     }
   }
 
+  /**
+   * Throws SqlTimeoutException if the statement has already timed out
+   * @throws AlreadyClosedSqlException   if Connection is closed
+   * @throws SqlTimeoutException         if Statement is already timed out
+   */
+  private void throwIfTimedOut() throws AlreadyClosedSqlException,
+                                        SqlTimeoutException,
+                                        SQLException {
+    if ( (statement instanceof DrillStatementImpl && ((DrillStatementImpl) statement).isTimedOut()) ||
```

Comment:
since PreparedStatement extends from Statement, maybe DrillPreparedStatement should extends DrillStatement? this way, you could add isTimedOut to the interface? (but you would not need to if done in DrillCursor ;-))